src/dmd/cparse.d

```diff
@@ -3445,6 +3451,25 @@ final class CParser(AST) : Parser!AST
         return stc;
     }
 
+    /***********************
+     * Return suitable integer type for the given size and signedness
+     * Params:
+     *  size = size of type
+     *  unsigned = true if unsigned
+     * Returns:
+     *  corresponding D type
+     */
+    private AST.Type integerTypeForSize(ubyte size, bool unsigned)
```

Comment:
Instead of the extra parameter, I suggest two functions would be simpler. Adding `unsignedTypeForSize` should do the trick.